parallel-cluster-post-install.sh

```diff
@@ -68,87 +88,271 @@ sed -i "s|@CUDA_INSTALL_PATH@|${CUDA_INSTALL_PATH}|g" /etc/profile.d/cuda.sh
+  chown -R ${OSUSER}:${OSGROUP} ${CRYOSPARC_INSTALL_PATH}/cryosparc_worker
+
+  # Start cryoSPARC main package 
+  /bin/su -c "${CRYOSPARC_INSTALL_PATH}/cryosparc_master/bin/cryosparcm stop" - ${OSUSER}
 
-# Install cryoSPARC work package
-cd "${CRYOSPARC_INSTALL_PATH}" || return
-tar -xf cryosparc_worker.tar.gz
-chown -R ${OSUSER}:${OSGROUP} /shared/cryosparc
-cd cryosparc_worker || return
-/bin/su -c "\"${CRYOSPARC_INSTALL_PATH}\"/cryosparc_worker/install.sh \
-    --license \"${CRYOSPARC_LICENSE_ID}\" \
-    --cudapath \"${CUDA_INSTALL_PATH}/${CUDA_VERSION}\" \
-    --yes" - ${OSUSER}
+else 
+  echo "Restoring CryoSPARC with updated Hostname and refreshing paritition connections"
 
-# Clean up
-rm "${CRYOSPARC_INSTALL_PATH}"/*.tar.gz
+  # Stop any running cryosparc
+  systemctl stop cryosparc-supervisor.service || /bin/su -c "${CRYOSPARC_INSTALL_PATH}/cryosparc_master/bin/cryosparcm stop || echo \"Nothing Running\" " - ${OSUSER}
 
+  # Update hostname to new main
+  sed -i "s/^\(.*CRYOSPARC_MASTER_HOSTNAME=\"\).*\"/\1$HOSTNAME\"/g" ${CRYOSPARC_INSTALL_PATH}/cryosparc_master/config.sh
+
+  # Once again, re-align permissions for proper start 
+  chown -R ${OSUSER}:${OSGROUP} ${CRYOSPARC_INSTALL_PATH}/cryosparc_master
+fi
+  
 # Start cluster
-/bin/su -c "${CRYOSPARC_INSTALL_PATH}/cryosparc_master/bin/cryosparcm start" - ${OSUSER}
+#/bin/su -c "${CRYOSPARC_INSTALL_PATH}/cryosparc_master/bin/cryosparcm start" - ${OSUSER}
+
+# Confirm Restart CryoSPARC main
+/bin/su -c "cd ${CRYOSPARC_INSTALL_PATH} && ${CRYOSPARC_INSTALL_PATH}/cryosparc_master/bin/cryosparcm restart" - ${OSUSER} 
+
+# Stop server in anticipation for service
+/bin/su -c "cd ${CRYOSPARC_INSTALL_PATH} && ${CRYOSPARC_INSTALL_PATH}/cryosparc_master/bin/cryosparcm stop" - ${OSUSER} 
 
+# Create the CryoSPARC Systemd service and start at Boot
+eval $(cryosparcm env)
+cd "${CRYOSPARC_INSTALL_PATH}/cryosparc_master/systemd" || return
+# Final alignment on permissions
+chown -R ${OSUSER}:${OSGROUP} ${CRYOSPARC_INSTALL_PATH}/cryosparc_master/systemd
+env "CRYOSPARC_ROOT_DIR=$CRYOSPARC_ROOT_DIR" ./install_services.sh
+systemctl start cryosparc-supervisor.service
+systemctl restart cryosparc-supervisor.service
+systemctl enable cryosparc-supervisor.service
+
+
+# Be tolerant of errors on partitions; we can always come back through admin panel and add later
+set +e 
+echo "Partitions:"
+/opt/slurm/bin/scontrol show partitions
+
+echo "Beginning"
 # Create cluster config files
-for PARTITION in gpu-t4 gpu-a100 gpu-v100 cpu gpu-a100-spot
-do
-    cat > "${CRYOSPARC_INSTALL_PATH}"/cluster_info.json << 'EOF'
+for PARTITION in $( /opt/slurm/bin/scontrol show partitions | grep PartitionName | cut -d'=' -f 2 )
+do 
+  if [ ! -f "${CRYOSPARC_INSTALL_PATH}/${PARTITION}/cluster_info.json" ]; then
+    echo "Connecting New Partition: ${PARTITION}"
+    case $PARTITION in
+    	gpu-t4*)
+	    echo "T4 GPU: $PARTITION"
+            PARTITION_CACHE_PATH="/scratch"
+            PARTITION_CACHE_RESERVE=10000
```

Comment:
This is consistent across all cases

Reply:
These are settings for each lane in CryoSPARC